x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -44,11 +61,157 @@ public LogicalPlan visitEventQuery(EqlBaseParser.EventQueryContext ctx) {
+                numberOfKeys = keySize;
+            } else {
+                if (numberOfKeys != keySize) {
+                    Source src = source(sequenceTermCtx.by != null ? sequenceTermCtx.by : sequenceTermCtx);
+                    int expected = numberOfKeys - parentJoinKeys.size();
+                    int found = keySize - parentJoinKeys.size();
+                    throw new ParsingException(src, "Inconsistent number of join keys specified; expected [{}] but found [{}]", expected,
+                            found);
+                }
+                queries.add(sequenceTerm);
+            }
+        }
+
+        return new Sequence(source(ctx), queries, until, maxSpan);
+    }
+
+    public KeyedFilter visitSequenceTerm(SequenceTermContext ctx, List<Expression> joinKeys) {
+        if (ctx.FORK() != null) {
+            throw new ParsingException(source(ctx.FORK()), "sequence fork is unsupported");
+        }
+
+        List<Expression> keys = CollectionUtils.combine(joinKeys, visitJoinKeys(ctx.by));
+        return new KeyedFilter(source(ctx), visitEventQuery(ctx.subquery().eventQuery()), keys);
+    }
+
+    @Override
+    public TimeValue visitSequenceParams(SequenceParamsContext ctx) {
+        if (ctx == null) {
+            return TimeValue.MINUS_ONE;
+        }
+
+        NumberContext numberCtx = ctx.timeUnit().number();
+        if (numberCtx instanceof IntegerLiteralContext) {
+            Number number = (Number) visitIntegerLiteral((IntegerLiteralContext) numberCtx).fold();
+            long value = number.longValue();
+            
+            if (value <= 0) {
+                throw new ParsingException(source(numberCtx), "A positive maxspan value is required; found [{}]", value);
+            }
+            
+            String timeString = text(ctx.timeUnit().IDENTIFIER());
+            TimeUnit timeUnit = TimeUnit.SECONDS;
+            if (timeString != null) {
+                switch (timeString) {
+                    case "":
+                    case "s":
+                    case "sec":
+                    case "secs":
+                    case "second":
+                    case "seconds":
+                        timeUnit = TimeUnit.SECONDS;
+                        break;
+                    case "m":
+                    case "min":
+                    case "mins":
+                    case "minute":
+                    case "minutes":
+                        timeUnit = TimeUnit.MINUTES;
+                        break;
+                    case "h":
+                    case "hs":
+                    case "hour":
+                    case "hours":
+                        timeUnit = TimeUnit.HOURS;
+                        break;
+                    case "d":
+                    case "ds":
+                    case "day":
+                    case "days":
+                        timeUnit = TimeUnit.DAYS;
+                        break;
+                    default:
```

Comment:
I added #54760 to track an issue for removing unknown time units and only supporting ones that are recognized by range queries.